doc/source/tutorial.rst

```diff
@@ -121,10 +123,10 @@ Create a RunEngine:
 This RunEngine is ready to use --- but if you care about visualizing or saving
 your data, there is more to do first....
 
-The RunEngine dispatches a live stream of metadata and data to one or more
-consumers ("callbacks") for in-line data processing and visualization and
-long-term storage. Example consumers include a live-updating plot, a
-curve-fitting algorithm, a database, a message queue, or a file in your
+During data acquisition, the RunEngine dispatches a live stream of metadata and
+data to one or more consumers ("callbacks") for in-line data processing and
+visualization and long-term storage. Example consumers include a live-updating
+plot, a curve-fitting algorithm, a database, a message queue, or a file in your
 preferred format.
```

Comment:
Maybe quick disclaimer discouraging the user to run complex analysis through this:

```py
NOTE : The RunEngine will block until these callbacks complete. If any heavy-weight analysis is needed, consider using the `zeroMQ callbacks`_ or executing your function in a separate thread.

.. _zeroMQ callbacks: http://nsls-ii.github.io/bluesky/callbacks.html?highlight=zmq#subscriptions-in-separate-processes-or-host-with-0mq
```

Reply:
Good idea. Given the broad audience for this document, which includes some visiting chemist just trying to `count`, I'm taking pains to lay on the detail too thick too soon. ("What's a thread? What's a zeroMQ? Do I need to know this to take my data?") Therefore, I have put this explanation at the top of `callbacks.rst` and I have added a link from here to `callbacks.rst`. One goal of this rewrite was to consolidate the less-technical documentation in `tutorial.rst` so that we can let loose with these important details in other sections, like `callbacks.rst`.